mos_tests/murano/actions.py

```diff
@@ -0,0 +1,226 @@
+        def is_murano_env_deployed():
+            status = self.murano.environments.get(environment.id).status
+            if status == 'deploy failure':
+                raise Exception('Environment deploy finished with errors')
+            return status == 'ready'
+
+        try:
+            wait(is_murano_env_deployed, timeout_seconds=1200,
+                 waiting_for='Environment is ready')
+        except Exception:
+            raise Exception('Environment deploy finished with errors')
+
+        environment = self.murano.environments.get(environment.id)
+        logs = self.get_log(environment)
+        assert 'Deployment finished' in logs
+        return environment
+
+    def deploy_environment(self, environment, session):
+        self.murano.sessions.deploy(environment.id, session.id)
+        return self.wait_for_deploy(environment)
+
+    def get_action_id(self, environment, name, service):
+        env_data = environment.to_dict()
+        a_dict = env_data['services'][service]['?']['_actions']
+        for action_id, action in a_dict.items():
+            if action['name'] == name:
+                return action_id
+
+    def run_action(self, environment, action_id):
+        self.murano.actions.call(environment.id, action_id)
+        return self.wait_for_deploy(environment)
+
+    def status_check(self, environment, configurations, kubernetes=False,
+                     negative=False):
+        for configuration in configurations:
+            if kubernetes:
+                service_name = configuration[0]
+                inst_name = configuration[1]
+                ports = configuration[2:]
+                ip = self.get_k8s_ip_by_instance_name(environment, inst_name,
+                                                      service_name)
+                if ip:
+                    for port in ports:
+                        assert self.check_port_access(ip, port, negative)
+                        assert self.check_k8s_deployment(ip, port, negative)
+                else:
+                    raise Exception("Instance {} doesn't have floating IP"
+                                    .format(inst_name))
+            else:
+                inst_name = configuration[0]
+                ports = configuration[1:]
+                ip = self.get_ip_by_instance_name(environment, inst_name)
+                if ip and ports:
+                    for port in ports:
+                        assert self.check_port_access(ip, port)
+                else:
+                    raise Exception("Instance {} doesn't have floating IP"
+                                    .format(inst_name))
+
+    def check_port_access(self, ip, port, negative=False):
+
+        def connect():
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            result = sock.connect_ex((str(ip), port))
+            sock.close()
+            return result == 0
+        try:
+            wait(connect, timeout_seconds=300,
```

Comment:
In negative case this code will wait for 5 minutes. Is this behavior is what you want?